### PR TITLE
Set short-message feature unstable

### LINKED ISSUE #45995: `--error-format=short` should be an unstable option
i.e., require `-Zunstable-features`. This should be backported to beta if necessary.

cc @GuillaumeGomez @alexcrichton @estebank

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #45995.

r? @nrc

### CODE REVIEW

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -1435,8 +1435,16 @@ pub fn build_session_options_and_crate_config(matches: &getopts::Matches)
...
             Some("pretty-json") => ErrorOutputType::Json(true),
-            Some("short") => ErrorOutputType::Short(color),
-
+            Some("short") => {
+                if nightly_options::is_unstable_enabled(matches) {
+                    ErrorOutputType::Short(color)
+                } else {
+                    early_error(error_format,
+                                &format!("the `-Z unstable-options` flag must also be passed to \
+                                          enable the target-spec-json print option"));
```
Comment (QuietMisdreavus) [inline src/librustc/session/config.rs]: is "target-spec-json" the right thing to say here?

Comment (GuillaumeGomez) [reply src/librustc/session/config.rs]: No. Good catch, thanks!

**Comment by QuietMisdreavus** [comment]
also:

```
[00:04:37] tidy error: /checkout/src/librustc/session/config.rs:1482: tab character
[00:04:37] tidy error: /checkout/src/librustc/session/config.rs:1482: trailing whitespace
[00:04:39] some tidy checks failed
```

**Comment by QuietMisdreavus** [comment]
```
[00:06:38] error[E0425]: cannot find value `error_format` in this scope
[00:06:38]     --> /checkout/src/librustc/session/config.rs:1478:33
[00:06:38]      |
[00:06:38] 1478 |                     early_error(error_format,
[00:06:38]      |                                 ^^^^^^^^^^^^ not found in this scope
[00:06:38] 
[00:06:57] error: aborting due to previous error
[00:06:57] 
[00:06:57] error: Could not compile `rustc`.
```

**Comment by GuillaumeGomez** [comment]
So inconvenient to not be able to compile the code haha.

**Comment by kennytm** [comment]
Please update `src/test/ui/short-error-format.rs` to include the `-Z unstable-options` flag.

**Comment by kennytm** [comment]
@GuillaumeGomez is this PR ready to be merged? If yes, please remove the “WIP” from the title.

**Comment by GuillaumeGomez** [comment]
No it isn't, sorry about that.

@bors: r=kennytm

**Comment by kennytm** [comment]
@GuillaumeGomez Since this is included in the rollup #46029, should that be closed or what

**Pull request merged**